integration/networking/resolvconf_test.go

```diff
@@ -9,30 +10,39 @@ import (
 	"github.com/docker/docker/integration/internal/container"
 	"github.com/docker/docker/integration/internal/network"
 	"github.com/docker/docker/testutil/daemon"
+	"github.com/miekg/dns"
 	"gotest.tools/v3/assert"
 	is "gotest.tools/v3/assert/cmp"
 	"gotest.tools/v3/skip"
 )
 
-// Regression test for https://github.com/moby/moby/issues/46968
-func TestResolvConfLocalhostIPv6(t *testing.T) {
-	// No "/etc/resolv.conf" on Windows.
-	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
-
-	ctx := setupTest(t)
-
-	// Write a resolv.conf that only contains a loopback address.
+// writeTempResolvConf writes a resolv.conf that only contains a single
+// nameserver line, with address addr.
+// It returns the name of the temp file.
+func writeTempResolvConf(t *testing.T, addr string) string {
+	t.Helper()
 	// Not using t.TempDir() here because in rootless mode, while the temporary
 	// directory gets mode 0777, it's a subdir of an 0700 directory owned by root.
 	// So, it's not accessible by the daemon.
 	f, err := os.CreateTemp("", "resolv.conf")
 	assert.NilError(t, err)
-	defer os.Remove(f.Name())
```

Comment:
Just because you aren't using t.TempDir() doesn't mean you have to give up auto-cleanup.
```go
t.Cleanup(func() { os.Remove(f.Name()) })
```

Reply:
Done.